test_project/store/src/main/java/test/sergeyvalyushko/store/Category.java

```diff
@@ -0,0 +1,14 @@
+package test.sergeyvalyushko.store;
+
+import test.sergeyvalyushko.store.helpers.RandomStorePopulator;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+public abstract class Category {
+    protected List<Product> productList = new ArrayList<Product>();
+    public abstract List<Product> addProduct(RandomStorePopulator populator);
```

Comment:
what for this method is abstract, cause it is the same in all implementations. The same note for getProductList method